cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -258,6 +259,39 @@ MapBuilderBridge::GetLocalTrajectoryData() {
   return local_trajectory_data;
 }
 
+void MapBuilderBridge::HandleTrajectoryQuery(
+    cartographer_ros_msgs::TrajectoryQuery::Request& request,
+    cartographer_ros_msgs::TrajectoryQuery::Response& response) {
+  const auto trajectory_states = GetTrajectoryStates();
+  if (!trajectory_states.count(request.trajectory_id) ||
+      trajectory_states.at(request.trajectory_id) ==
```

Comment:
please, use `find` and the resulting iterator instead of searching twice.

Reply:
I will use the new [`TrajectoryStateToStatus`](https://github.com/googlecartographer/cartographer_ros/commit/5fb4314c8b7f235af121c2bce2a5761e0adb8651#diff-e1f4eea55ec120f9e563c2b5d947449aR483) here and do the `find` optimization in `TrajectoryStateToStatus`.